microcosm_metrics/factories.py

```diff
@@ -0,0 +1,78 @@
+"""
+Factories for statsd/metrics clients.
+
+"""
+from types import MethodType
+
+from datadog import DogStatsd
+from microcosm.api import defaults
+from statsd import StatsClient
+
+
+def attach(graph, statsd):
+    """
+    Attach statsd client to `metrics` key for pseudo polymorphism.
+
+    """
+    graph.metrics = statsd
+    return statsd
+
+
+@defaults(
+    host="localhost",
+    port=8125,
+    tags=[],
+)
+def configure_statsd(graph):
+    """
+    Create a vanilla statsd client.
+
+    """
+    if graph.metadata.testing:
+        from mock import MagicMock
+        cls = MagicMock
+    else:
+        cls = StatsClient
+
+    statsd = cls(
+        host=graph.config.datadog_statsd.host,
+        port=graph.config.datadog_statsd.port,
+    )
+
+    if not graph.metadata.testing:
+        # We have different interfaces between the StatsClient and DogStatsd. Try to adapt the
```

Comment:
Ugliness. The two statsd clients do not have the same interface. I'm adapting the vanilla client to the datadog interface here, at least for the parts we use, because datadog has a better Python API.

This could be much cleaner.

It also doesn't work for `histogram` (there's no histogram support in the Python statsd client).